src/coreclr/vm/classlayoutinfo.cpp

```diff
@@ -148,6 +148,9 @@ namespace
             LayoutRawFieldInfo* pfwalk = *pSortWalk;
             RawFieldPlacementInfo* placementInfo = &pfwalk->m_placement;
 
+            if (placementInfo->m_size > FIELD_OFFSET_LAST_REAL_OFFSET)
```

Comment:
Does this also cover the case where we have a struct with many fields (each field is less than `FIELD_OFFSET_LAST_REAL_OFFSET` in size) and the size of all fields together overflows Int32?

Reply:
No, I could do that. the thing is `FIELD_OFFSET_LAST_REAL_OFFSET` is smaller than `int.MaxValue`. So this would error out too.
```c#
[StructLayout(LayoutKind.Sequential, Size = int.MaxValue)] 
struct BigArray
{
}
```
Is that expected? I wasn't sure.

Reply:
This should continue to work. There is no reason to break that.

Note that `struct X { BigArray a; byte x; }` fails to load today (that's correct). It is only `struct X { byte x; BigArray a; }` that loads successfully. It suggests that we do have the right checks that prevent overflows of field offsets in place, but we are missing a check that would catch overflow of the total size.

Reply:
That's interesting. I will look into it.

Reply:
@jkotas Should we allow this or not?
```
[StructLayout(LayoutKind.Sequential, Size = int.MaxValue)] 
        struct BigArray
        {
        }
```